torch/csrc/jit/tensorexpr/ir_simplifier.cpp

```diff
@@ -1663,9 +1663,9 @@ const Expr* polyGCD(const Polynomial* poly) {
   // We ony want to factorize if we're saving complete operations, i.e. no
   // value in factorizing 6x + 4y into 2 * (3x + 2y) since we don't save work.
   int opsSaved = 1; // default to saving the scalar.
-  long GCD = immediateAs<long>(scalar);
+  long GCD = std::abs(immediateAs<long>(scalar));
```

Comment:
Is this correct if both LHS and RHS are negative? Do we have a test of that case?

Reply:
Even if both LHS and RHS are negative, GCD should still be a positive integer, AFAIK. Do you think there is a problem with that?

Reply:
No I think thats correct, I just like to be as comprehensive with tests as its practical to be. NBD.

Reply:
Sure, sounds good.